crates/server/src/proxy/compute/data_collection/components/convert.rs

```
@@ -155,3 +159,50 @@ fn convert_dict(dict: Option<HashMap<String, serde_json::Value>>) -> Vec<(String
         })
         .collect()
 }
+
+fn convert_products(
+    properties: Option<HashMap<String, serde_json::Value>>,
+) -> Vec<Vec<(String, String)>> {
+    use serde_json::Value;
+
+    let Some(dict) = properties else {
+        return Vec::new();
+    };
+    
+
+    // if the key is products, then we need to convert the value to a list of tuples
+    if let Some(products) = dict.get("products") {
+        // if products is not an array, return an empty vector
+        if !products.is_array() {
+            return Vec::new();
```

should we log this somehow so that we can more easily debug this scenario? if `products` is meant to always be an array, we should make sure the developer has a way to notice that mistake instead of silently failing